TAO/tests/Oneway_Send_Timeouts/Server_Task.h

```diff
@@ -3,34 +3,32 @@
 
 #include "Server.h"
 
-#include "ace/Task.h"
 #include "ace/ARGV.h"
-#include "ace/Auto_Ptr.h"
+#include "ace/Task.h"
+
+#include <string>
 
 class Server_Task : public ACE_Task_Base
 {
- public:
-
+public:
   Server_Task (const std::string& args)
     : args_ (args)
-  { }
-  ~Server_Task ()
-    {
-      this->force_shutdown ();
-    }
+  {}
+  ~Server_Task () { this->force_shutdown (); }
 
   virtual int svc ()
   {
     bool initializer = false;
     {
       ACE_GUARD_RETURN (TAO_SYNCH_MUTEX, ace_mon, this->mutex_, -1);
-      if (server_.get() == 0) {
-  ACE_ARGV my_args (args_.c_str());
+      if (server_.get () == 0) {
+        ACE_ARGV my_args (args_.c_str ());
 
-  // Initialize Server ORB in new thread
-  ACE_auto_ptr_reset (server_, new Server (my_args.argc(), my_args.argv()));
-  ACE_ASSERT (server_.get() != 0);
-  initializer = true;
+        // Initialize Server ORB in new thread
+        // server_ = std::make_unique<Server>(my_args.argc(), my_args.argv());
```

Comment:
You could check `ACE_HAS_CPP14` and use at that moment `make_unique`, if not set, use new. That way we can easily modernize the code at the moment we migrate to C++14

Reply:
I do not understand why here 2 args are used?

```c++
server_ = std::make_unique<Server>(my_args);
```
Should work, or not?

Reply:
OK